File: src/main/java/com/szokone/aem/gokotlin/java/service/ConfigurableService.java
package com.szokone.aem.gokotlin.java.service;

import java.util.Map;

/**
 * Service Interface
 */
public interface ConfigurableService {

    /**
     * @return true if enabled
     */
    boolean isEnabled();

    /**
     * @return map of text in component to replace
     */
    Map<String, String> getReplacements();

}

File: src/main/kotlin/com/szokone/aem/gokotlin/kotlin/service/ConfigurableService.kt
package com.szokone.aem.gokotlin.kotlin.service

/**
 * Service Interface
 */
interface ConfigurableService {

    /**
     * @return true if enabled
     */
    fun isEnabled(): Boolean

    /**
     * @return list of HTML tags that should be decorated with additional classes
     */
    fun getDecorableHtmlTags(): Array<String>

    /**
     * @return additional classes that will be added to 'decorableHtmlTags'
     */
    fun getAdditionalClasses(): String
}